src/opm/input/eclipse/EclipseState/InitConfig/Equil.cpp

```diff
@@ -93,60 +101,136 @@ namespace Opm {
+        return this->datum_depth;
+    }
+
+    double StressEquilRecord::datumPosX() const {
+        return this->datum_posx;
+    }
+
+    double StressEquilRecord::datumPosY() const {
+        return this->datum_posy;
+    }
+
+    double StressEquilRecord::stressXX() const {
+        return this->stress_xx;
+    }
+
+    double StressEquilRecord::stressXX_grad() const {
+        return this->stress_xx_grad;
+    }
+
+    double StressEquilRecord::stressYY() const {
+        return this->stress_yy;
+    }
+
+    double StressEquilRecord::stressYY_grad() const {
+        return this->stress_yy_grad;
+    }
+
+    double StressEquilRecord::stressZZ() const {
+        return this->stress_zz;
+    }
+
+    double StressEquilRecord::stressZZ_grad() const {
+        return this->stress_zz_grad;
+    }
+
+    bool StressEquilRecord::operator==(const StressEquilRecord& data) const {
+        return datum_depth == data.datum_depth &&
+               datum_posx == data.datum_posx &&
+               datum_posy == data.datum_posy &&
+               stress_xx == data.stress_xx &&
+               stress_xx_grad == data.stress_xx_grad &&
+               stress_yy == data.stress_yy &&
+               stress_yy_grad == data.stress_yy_grad &&
+               stress_zz == data.stress_zz &&
+               stress_zz_grad == data.stress_zz_grad;
+    }
+
     /* ----------------------------------------------------------------- */
 
-    Equil::Equil( const DeckKeyword& keyword )
+    template<class RecordType>
+    EquilContainer<RecordType>::EquilContainer(const DeckKeyword& keyword)
     {
         using ParserKeywords::EQUIL;
 
         for (const auto& record : keyword) {
-            const auto datum_depth_arg = record.getItem<EQUIL::DATUM_DEPTH>().getSIDouble(0);
-            const auto datum_depth_pc_arg = record.getItem<EQUIL::DATUM_PRESSURE>().getSIDouble(0);
-            const auto woc_depth = record.getItem<EQUIL::OWC>().getSIDouble(0);
-            const auto woc_pc = record.getItem<EQUIL::PC_OWC>().getSIDouble(0);
-            const auto goc_depth = record.getItem<EQUIL::GOC>().getSIDouble(0);
-            const auto goc_pc = record.getItem<EQUIL::PC_GOC>().getSIDouble(0);
-            const auto live_oil_init = record.getItem<EQUIL::BLACK_OIL_INIT>().get<int>(0) <= 0;
-            const auto wet_gas_init = record.getItem<EQUIL::BLACK_OIL_INIT_WG>().get<int>(0) <= 0;
-            const auto target_accuracy = record.getItem<EQUIL::OIP_INIT>().get<int>(0);
-            const auto humid_gas_init = record.getItem<EQUIL::BLACK_OIL_INIT_HG>().get<int>(0) <= 0;
-
-            this->m_records.emplace_back(datum_depth_arg, datum_depth_pc_arg,
-                                         woc_depth      , woc_pc,
-                                         goc_depth      , goc_pc,
-                                         live_oil_init, wet_gas_init, target_accuracy, humid_gas_init);
+            this->m_records.emplace_back(RecordType{record});
```

Comment:
If you're mentioning the type and calling the constructor anyway, you should be using `push_back()` instead.  `Emplace_back()` is for in-place construction and calling the constructor directly defeats that purpose.

Reply:
right you are, my bad, fixed. since you want it explicit I use push_back, not emplace_back(record).

Reply:
> since you want it explicit I use push_back, not emplace_back(record).

No objection, but `vector<T>::emplace_back()` is supposed to be able to call explicit constructors for `T`.  Did you get a build failure with `emplace_back()`?

Reply:
right you are, as usual. nope. that's better. let's do that.